### ISSUE TITLE
[Bug?] All torrents has 1/1/1970 as added date.

### ISSUE THREAD

**Issue by NoLooseEnds** [jfurrow/flood#338]
I run it on an Ubuntu 10.16, torrents and files are stored on an NFS share.

**Comment by jfurrow** [comment]
Just to be 100% sure, you are adding them via the Flood UI, right? If so, how are you adding them (files or URLs)?

**Comment by Wonderfall** [comment]
I don't have this problem when I adds torrents via the Flood UI, but using scgi directly I also have this issue (not sure if it's related to Flood UI, right?).

**Comment by nVitius** [comment]
@Wonderfall That would be correct. Flood adds a custom property for the date.

**Comment by NoLooseEnds** [comment]
Not adding them through Flood, but automagically via Flexget.

Is there a way to let Flood read the creation date of the torrent file (or pull the data from rtorrent when it started the torrent).

**Comment by frebib** [comment]
I can confirm this same bug when torrents are added through sickrage

**Comment by wopian** [comment]
Had this happen many times with magnets added via Flood

**Comment by jfurrow** [comment]
I agree the date field should display `Unknown` if it's undefined, so I'll use this issue for that.

To reiterate: Flood adds a custom attribute on the torrent when the torrent is added (the same custom attribute that ruTorrent uses). So adding torrents via any other method will not have this attribute.

**Comment by NoLooseEnds** [comment]
I am pretty sure rutorrent showed the added date and the completed date even if torrents added outside of rutorrent.

(Source: I've used rutorrent in combination with flexget for years.)

**Comment by jfurrow** [comment]
As far as I know, Flood uses the same custom attribute as ruTorrent (it's `addtime`). Maybe ruTorrent looks for other custom properties as well... I'll look into this.

**Comment by liquidox** [comment]
One of the reasons I am looking into Flood is the problem with unknown dates in rutorrent. Torrents I add through Remote Torrent Added or Sonarr mostly show up with 1/1/1970. This makes sorting by date pointless. I suppose this is just a problem in rtorrent? Perhaps a new column could be added that shows the date the torrent appeared in Flood, then we could just sort on that column instead of the broken one.

**Comment by gdob** [comment]
> Torrents I add through Remote Torrent Added or Sonarr mostly show up with 1/1/1970.

Ohh, I thought I had something wrong in my setup, but I see others have the same problem. It works fine when you add a torrent through the Fluid UI, but if the torrent is added via Sonarr or Radarr the date shown is 1/1/1970 for some reason. Older torrents added while I was using ruTorrent show the correct date on Fluid.

**Comment by spiderknight** [comment]
This issue also exists when adding torrents through the rwatch folder. ruTorrent does not have this problem with rwatch/.

**Comment by bserrao** [comment]
I can confirm the same behavior when using watch folders.

**Comment by joydashy** [comment]
This could be solved by a Flexget Output plugin for Flood, anyone know if this exists? I can't find it through Googling...

**Comment by Sparticuz** [comment]
Try adding this to your rtorrent.rc
```
# Method to add the start time of the torrent
method.set_key = event.download.inserted_new, !time_stamp, "d.custom.set=addtime,$cat=$system.time= ; d.save_resume="
```
